Framework/Algorithms/test/CorelliPowderCalibrationLoadTest.h

```diff
@@ -0,0 +1,117 @@
+    IAlgorithm_sptr lei =
+        AlgorithmFactory::Instance().create("LoadEmptyInstrument", 1);
+    lei->initialize();
+    lei->setPropertyValue("Filename", "NOW4_Definition.xml");
+    lei->setPropertyValue("OutputWorkspace", "wrongTypeWs");
+    lei->setPropertyValue("MakeEventWorkspace", "1");
+    lei->execute();
+    EventWorkspace_sptr ws =
+        AnalysisDataService::Instance().retrieveWS<EventWorkspace>(
+            "wrongTypeWs");
+    ws->mutableRun().addProperty<std::string>("start_time",
+                                              "2020-11-17T12:57:17", "", true);
+
+    // setup alg
+    CorelliPowderCalibrationLoad alg;
+    alg.initialize();
+    alg.setPropertyValue("InputWorkspace", "wrongTypeWs");
+    alg.setPropertyValue("DatabaseDir", ".");
+    alg.setPropertyValue("OutputWorkspace", "outWS");
+
+    // ensure that an exception is thrown here
+    TS_ASSERT_THROWS(alg.execute(), const std::runtime_error &);
+  }
+
+  void testValidateWSTime() {
+    // generate a mock workspace with correct instrument name
+    IAlgorithm_sptr lei =
+        AlgorithmFactory::Instance().create("LoadEmptyInstrument", 1);
+    lei->initialize();
+    lei->setPropertyValue("Filename", "CORELLI_Definition.xml");
+    lei->setPropertyValue("OutputWorkspace", "correctTypeWs");
+    lei->setPropertyValue("MakeEventWorkspace", "1");
+    lei->execute();
+    EventWorkspace_sptr ws =
+        AnalysisDataService::Instance().retrieveWS<EventWorkspace>(
+            "correctTypeWs");
+
+    // setup alg
+    CorelliPowderCalibrationLoad alg;
+    alg.initialize();
+    alg.setPropertyValue("InputWorkspace", "correctTypeWs");
+    alg.setPropertyValue("DatabaseDir", ".");
+    alg.setPropertyValue("OutputWorkspace", "outWS");
+
+    // ensure that an exception is thrown here
+    TS_ASSERT_THROWS(alg.execute(), const std::runtime_error &);
+  }
+
+  void testExec() {
+    // generate a mock workspace with correct instrument name
+    IAlgorithm_sptr lei =
+        AlgorithmFactory::Instance().create("LoadEmptyInstrument", 1);
+    lei->initialize();
+    lei->setPropertyValue("Filename", "CORELLI_Definition.xml");
+    lei->setPropertyValue("OutputWorkspace", "correctTypeWs");
+    lei->setPropertyValue("MakeEventWorkspace", "1");
+    lei->execute();
+    EventWorkspace_sptr ws =
+        AnalysisDataService::Instance().retrieveWS<EventWorkspace>(
+            "correctTypeWs");
+
+    // add a starting time to log
+    ws->mutableRun().addProperty<std::string>("start_time",
+                                              "2020-11-17T12:57:17", "", true);
+
+    // setup alg
+    CorelliPowderCalibrationLoad alg;
+    alg.initialize();
+    alg.setPropertyValue("InputWorkspace", "correctTypeWs");
+    alg.setPropertyValue("DatabaseDir", ".");
+    alg.setPropertyValue("OutputWorkspace", "outWS");
+
+    // make sure no exception is thrown here
+    TS_ASSERT_THROWS_NOTHING(alg.execute())
```

Comment:
I guess I must be missing something because I don't understand why `CorelliPowderCalibrationLoad` doesn't throw in the absence of a calibration file. Can you please clarify?

Reply:
My understanding is that the `TS_ASSERT_*` did not catch the file not found error from the child algorithm (LoadAscii), which is kind of puzzling for me as well.  I will ask around to see if this is a intended feature.

Reply:
Running the manual test in the workbench, in the absence of a calibration file, will throw a `ValueError`. I'm surprised the exception is not picked up by the test.

Reply:
An update on this part (also available as comments in the source code in the upcoming commit):

- The child alg `LoadAscii` did thrown an error due to not finding the file, but the CxxTest framework could not catch it at the Test level for this loader.
- Due to the error from the child alg, the main alg, a.k.a. the loader, did not finish, which is why a new check is inserted here `TS_ASSERT(!alg.isExecuted());` to reflect it.

Given that we don't understand why `TS_ASSERT_THROWS_NOTHING(alg.execute());` could not catch the error thrown by child alg, I'm keeping it in the UnitTest as a reminder in case we need to revisit this non-critical issue at a later date.